lib/loader/index.js

```diff
@@ -135,6 +186,16 @@ module.exports = {
       var firstScript = getFirstScript();
 
       firstScript.parentNode.insertBefore(script, firstScript);
+
+      _loadedUrls[url] = true;
+
+      if (options.namespaceName) {
+        namespace.loadedUrls = _loadedUrls;
```

Comment:
I've never used [`set`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/set) before, could we use it to enforce this? Could you determine if the property owner is the one making a modification by inspecting the context?